stdlib/asm/crypto/fri.masm

```diff
@@ -0,0 +1,607 @@
+#! in https://github.com/0xPolygonMiden/miden-vm/issues/592#issue-1481983459
+#!
+#! Expected stack state
+#!
+#! [_, _, τ1, τ0, mem_ptr_p, ...]
+#!
+#! Final stack state
+#!
+#! [α1, α0, _, _, ...]
+#!
+#! Note, mem_ptr_p is absolute memory address of polynomial p ( which was interpolated using advice provider )
+#! s.t. next 3 decreasing ( by 1 ) memory addresses hold remaining 6 coefficients of p. Why this descending order of
+#! memory addressing ? - Read point 2 of https://github.com/0xPolygonMiden/miden-vm/issues/592#issue-1481983459
+proc.compute_alpha
+    movup.4
+    dup
+    sub.1
+    movdn.5
+    push.0.0.0.0
+    movup.4
+    mem_loadw
+
+    dup.7
+    dup.7
+    ext2mul
+    ext2add
+
+    movup.6
+    dup
+    sub.1
+    movdn.7
+    push.0.0.0.0
+    movup.4
+    mem_loadw
+
+    movup.5
+    movup.5
+    dup.9
+    dup.9
+    ext2mul
+    ext2add
+
+    dup.7
+    dup.7
+    ext2mul
+    ext2add
+
+    movup.6
+    dup
+    sub.1
+    movdn.7
+    push.0.0.0.0
+    movup.4
+    mem_loadw
+
+    movup.5
+    movup.5
+    dup.9
+    dup.9
+    ext2mul
+    ext2add
+
+    dup.7
+    dup.7
+    ext2mul
+    ext2add
+
+    movup.6
+    push.0.0.0.0
+    movup.4
+    mem_loadw
+
+    movup.5
+    movup.5
+    dup.9
+    dup.9
+    ext2mul
+    ext2add
+
+    movup.7
+    movup.7
+    ext2mul
+    ext2add
+end
+
+#! Given remainder codeword with 64 evaluations ( as starting memory address on stack ), this routine attempts to
+#! probabilistically verify remainder following the technique described in https://github.com/0xPolygonMiden/miden-vm/issues/592
+#! and an optimization of how to efficiently compute β in https://github.com/0xPolygonMiden/miden-vm/pull/644#pullrequestreview-1265141094
+#!
+#! Expected stack state
+#!
+#! [mem_ptr, ...]
+#!
+#! A few assumptions about mem_ptr
+#!
+#! - mem_ptr should be absolute memory address of the beginning of remainder codeword
+#! - Each evaluation is 2 field elements wide because they belong to quadratic extension field over F_p | p = 2^64 - 2^32 + 1
+#! - Meaning each memory address will hold two consecutive evaluations
+#! - Words ( four field elements ), in memory, are laid out in this order (a1_0, a1_0, a0_1, a0_0)
+#! - That means (a0_1, a0_0) -> first evaluation and (a1_1, a1_0) -> next evaluation
+#! - Next 31 memory addresses should be holding remaining 62 evaluations
+#! - If mem_ptr holds (a1_0, a1_0, a0_1, a0_0), then mem_ptr + 1, must hold (a3_0, a3_0, a2_1, a2_0)
+#! - That way, mem_ptr + 31 should be holding (a63_0, a63_0, a62_1, a62_0)
+#!
+#! Final stack looks like
+#!
+#! [...]
+#!
+#! If remainder verification fails, execution of the program stops.
+export.verify_remainder_64.4
+    locaddr.0
+    swap
+
+    exec.compute_tau
+    exec.compute_beta
+
+    movup.4
+    drop
+    locaddr.3
+    movdn.4
+
+    exec.compute_alpha
+
+    # assert α == β
+    movup.2
+    eq
+    movdn.2
+    eq
+    eq
+    assert
```

Comment:
As is, I think this could result in an error. Specifically, if both limbs of $\alpha$ and $\beta$ are different from each other ( $\alpha_0 \neq \beta_0$ and $\alpha_1 \neq \beta_1$) this will still pass. I think the code that we want here should be:
```
# assert α == β
movup.2
eq
movdn.2
eq
and
assert
```
Also, it would be helpful to document that the stack state before we enter this code block is:
```
[a1, a0, b1, b0, ... ]
```

Reply:
True, thanks for catching this.